src/main/java/com/intershop/version/semantic/VersionImpl.java

```diff
@@ -0,0 +1,167 @@
+package com.intershop.version.semantic;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Straight forward implementation of semantic versions without version extensions
+ */
+class VersionImpl implements SemanticVersion
```

Comment:
The naming seems to be a little weird: the interface name sounds more specific than its implementation. Maybe rename it.